### ISSUE TITLE
[joss-reviews#2974] Checkpoints used inconsistently

### ISSUE THREAD

**Issue by sjvrijn** [moead-framework/framework#64]
Related: openjournals/joss-reviews#2974 and #15

See e.g. the difference between 
https://github.com/moead-framework/framework/blob/659cb0404b356f80a30da83e63ff443104e4f2f2/moead_framework/algorithm/abstract_moead.py#L103-L104
and 
https://github.com/moead-framework/framework/blob/659cb0404b356f80a30da83e63ff443104e4f2f2/moead_framework/algorithm/combinatorial/moead_dra.py#L69-L70

**Comment by sjvrijn** [comment]
Instead of the externally referenced version given in the [tutorial section](https://moead-framework.github.io/framework/html/tuto.html?highlight=checkpoint#save-data-with-the-framework), it is probably better to have a checkpoint function always accept the MOEAD algorithm object as only argument. This would allow for a generic checkpoint function to be written without it being tied to a single MOEAD instance. I.e.
```python
def checkpoint(moead_algorithm: AbstractMoead):
    ....  # perform checkpoint actions here
```
with the call then becoming
```python
if checkpoint: 
    checkpoint(self)
```

A simple checkpoint function could then for example be added to the `tools` submodule as an example.

**Comment by geoffreyp** [comment]
The documentation and the code are updated in the PR #55 (commit 4717ec3)

**Comment by sjvrijn** [comment]
This issue can be closed when PR #55 is merged
